### DISCUSSION TITLE
Understanding the Sender Id vs Sender Party Id vs Receiver Id

### DISCUSSION THREAD

**Discussion by Upendra0** [phax/phase4#234] [Q&A] [answered]
Hi @all  and @phax ,
In phase4 spring boot demo application, During building the as4 message, the following code snipped are being used,
``` 
aBuilder = Phase4PeppolSender.builder ()
                    .documentTypeID (Phase4PeppolSender.IF.createDocumentTypeIdentifierWithDefaultScheme (docTypeId))
                    .processID (Phase4PeppolSender.IF.createProcessIdentifierWithDefaultScheme (processId))
                    .senderParticipantID (Phase4PeppolSender.IF.createParticipantIdentifierWithDefaultScheme (senderId)) // 1
                    .receiverParticipantID (aReceiverID)   // 2
                    .senderPartyID (seatId)                      // 3
                    .countryC1 (countryC1)
                    .payload (aDoc.getDocumentElement ())
                    .smpClient (aSMPClient)
                    .rawResponseConsumer (new AS4RawResponseConsumerWriteToFile())
 ```
 I highlighted the 3 fields, which are:

1.  Sender Participant Id (taking from controller)
2. Receiving Participant Id (taking from controller)
3. Sender Party Id (Reading seat id from configuration file).

Can you explain the value for this id by using reference of Peppol 4 corner model.
I am thinking as of now that:
1. Sender ID = ID of C1
2.  Receiver ID = ID of C4
3.  Sender Party Id = ID of C2

Please clarify if i got some wrong assumptions?

**Comment by phax** [comment, marked as answer]
@Upendra0 Thanks for reaching out to me.
Your assumptions are totally correct.
C1 + C4 are identified using a Peppol Participant ID
C2 + C3 are currently identified using the Peppol Seat ID (3 letters + 6 digits)

hth

**Reply by Upendra0** [reply]
@phax Thanks for quick response. It would be very great if you can clear another doubt regarding information stored in  SMP.
I am using PhossSMP for development and trying to understand the concept.

I seen in the Spring Boot Application code which is sending the message that  receiver ID (ID of C4) is being used for performing the SMP query to get the MetaData.

My question is ,  SMP contain the information of C4 or C3?
 When we create a new Service Group, we are  giving the ID of C4 in the ServiceGroup, which we are calling as receiver ID.
Or we are registering C3 in SMP which is AP of C4.

**Reply by phax** [reply]
Each participant managed in an SMP is currently a C4, an End User.
However, the connection data (AP URL and Certificate) point to the AP of C3.
That's how the connection between C3 and C4 is created.

**Reply by Upendra0** [reply]
Thank You for the quick response. It cleared my confusion regarding the connection of C3 and C4.
